### PR TITLE
feat(*): support kong lb

### PR DESCRIPTION
**Pull request by angao**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @your-reviewer

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/proxy/kong/kong.go [hunk 1]
```diff
@@ -230,7 +230,7 @@ func (k *kong) sync(lb *lbapi.LoadBalancer, dps []*appsv1.Deployment) error {
 		return err
 	}
 
-	return k.syncStatus(lb)
+	return k.syncStatus(lb, desiredDeploy.Name)
```
Comment (hicaistar) [inline pkg/proxy/kong/kong.go:233]: 使用 desiredDeploy := k.generateDeployment(lb)  这种方式获取而非传进去？WDYT？

Comment (angao) [reply pkg/proxy/kong/kong.go:233]: 为啥要调用两遍呢，`k.generateDeployment(lb)` 在上面不是被调用了么，而 deployment name 是不变的

Comment (hicaistar) [reply pkg/proxy/kong/kong.go:233]: 没啥大问题

Comment (angao) [reply pkg/proxy/kong/kong.go:233]: deployment name 是动态生成的，调用两遍 name 是不一样的。。

Comment (hicaistar) [reply pkg/proxy/kong/kong.go:233]: 大意了。👍

**Comment by hicaistar** [comment]
CI failed

于2020年10月27日星期二 15:11 <notifications@github.com> 写道:

*@angao* commented on this pull request.
------------------------------

In pkg/proxy/kong/kong.go
<https://github.com/caicloud/loadbalancer-controller/pull/196#discussion_r512458051>
:

> @@ -230,7 +230,7 @@ func (k *kong) sync(lb *lbapi.LoadBalancer, dps []*appsv1.Deployment) error {

 		return err

 	}



-	return k.syncStatus(lb)

+	return k.syncStatus(lb, desiredDeploy.Name)


deployment name 是动态生成的，调用两遍 name 是不一样的。。

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/caicloud/loadbalancer-controller/pull/196#discussion_r512458051>,
or unsubscribe

**Pull request merged**